tools/c7n_azure/c7n_azure/dependency_manager.py

```diff
@@ -41,8 +41,15 @@ def get_dependency_packages_list(packages, excluded_packages):
         res = [t for t in res if not any((e in t) for e in excluded_packages + packages)]
 
         # boto3 is a dependency for both c7n and c7n_mailer.. Remove the duplicate from the list
-        # because not all versions of pip can handle this.
-        res = [t for t in res if not any((e != t and t in e) for e in res)]
+        # because not all versions of pip can handle this.   
+        # use regex to get rid of duplicates excluding version number
+        regex = "^[^<>~=]*"
+        for i, val in enumerate(res):
+            pname = re.match(regex, val)
+            if sum(pname[0] in e for e in res) > 1:

```

Comment:
I would add .lower() for pname[0] and e to remove duplicates with case differences.

Reply:
Good catch. I will add it.